python/single_cell_model.cpp

```diff
@@ -270,6 +270,7 @@ void register_single_cell(pybind11::module& m) {
                 return m.traces();},
             "Holds sample traces after a call to run().")
         .def_readwrite("properties", &single_cell_model::gprop, "Global properties.")
+        .def_readwrite("catalogue", &single_cell_model::cat, "Mechanism catalogue.")
```

Comment:
The global properties hold a pointer to a catalogue.
At the moment this pointer is only set to point to the `single_cell_model::cat` when `::run` is called, so there will be a mismatch with the catalogue in the properties.

Instead, can users just set the catalogue by accessing `properties`, i.e:
```Python
model.properties.catalogue = ...;
```

Reply:
Yes, that's true. But we don't expose the `catalogue` of the global properties to the user any more. So `model.properties.catalogue` won't work. Instead we have `model.properties.register(cat)` which will point the global properties catalogue to `cat`.